### DISCUSSION TITLE
Too many badly named CanvasItem draw_ methods.

### DISCUSSION THREAD

**Discussion by setanarut** [godotengine/godot-proposals#4168] [Engine Core]
Too many badly named [CanvasItem draw_](https://docs.godotengine.org/en/latest/classes/class_canvasitem.html#methods) methods in Godot 4. There is no difference between `poly` and `multi`
A polyline can be opened/closed also a multiline can be opened/closed also A polygon can be opened/closed.

- `draw_multimesh()`
- `draw_line()`
- `draw_multiline()`
- `draw_multiline_colors()`
- `draw_polyline()`
- `draw_polyline_colors()`
- `draw_polygon()`
- `draw_colored_polygon()`
- `draw_primitive()`

>No practical difference in multi- vs. poly- prefixes. They're based on the word's Latin (multi-) or Greek (poly-) roots. The meaning difference is on the word rather than the prefixes in most cases. [[*]](https://www.quora.com/What-is-the-difference-between-multi-and-poly)

---

Another oddity. Primitives in geometry can have more than 4 vertices. But according to the `draw_primitive()` method this is not possible. 

---

There is no difference between a multi-line and a single-line if there are only two points.

## Proposal:

Reduced methods and more understandable names. Primitives such as `draw_circle()` and `draw_rect()` can remain.

- `draw_polymesh()`
- `draw_lines()` _# if there are two points it's a single line (simple and elegant)_
- `draw_regular_polygon(... closed: bool, fill: bool)` _# minimum 3 corner_
- `draw_polygon(... closed: bool, fill: bool)`

**Comment by setanarut** [comment]
Godot's terminology is really arbitrary and tedious. and unfortunately not simple and elegant

**Comment by Mickeon** [comment]
Good thing part of this has been brought up.
https://github.com/godotengine/godot/issues/54161#issuecomment-983846771

I suppose it would be a good idea to share some of the proposed renames in the Tracker itself, too.

**Reply by setanarut** [reply]
I copied it there, thanks.
